### PR TITLE
Auto-generate API schema for FE and cross-check on PR

### PR DESCRIPTION
**Pull request by will-sonnex**

### CODE REVIEW

File: packages/frontend/src/schema.ts [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+      criminalRecordCheckRequired: boolean;
+      idealVolunteer: string;
+      additionalInformation: string | null;
+    };
+  };
+}
+
+export interface operations {
+  OpportunityController_findAll: {
+    parameters: {};
```
Comment (stabback) [inline packages/frontend/src/schema.ts:99]: Based on the design, we will need:

1. Filter by user (or more simply, filter as "mine")
2. Filter by keyword
3. Sort

File: packages/frontend/src/schema.ts [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+    responses: {
+      default: {
+        content: {
+          "application/json": components["schemas"]["OpportunitySummaryResponseDto"][];
```
Comment (stabback) [inline packages/frontend/src/schema.ts:103]: We'll need a pagination spec to be able to wrap around this

File: packages/frontend/src/schema.ts (hunk 1 above)
Comment (stabback) [reply packages/frontend/src/schema.ts:99]: Unsure where you document headers, but we'll also need the auth header on this an all other endpoints

```
Authorization: Bearer {token}
```

File: packages/frontend/src/schema.ts [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+    OpportunityCreateDto: {
+      title: string;
+      contactName: string;
+      requiredPeopleCount: number;
+      /** @example 1666666666 */
+      startTime: number;
+      /** @example 1666666666 */
+      endTime: number;
+      /** @example 1666666666 */
+      postedTime: number;
```
Comment (stabback) [inline packages/frontend/src/schema.ts:41]: Should this be inferred instead of supplied?

Comment (will-sonnex) [reply packages/frontend/src/schema.ts:41]: Ah sorry yeah this shouldn't be in the create dto

File: packages/frontend/src/schema.ts (hunk 1 above)
Comment (will-sonnex) [reply packages/frontend/src/schema.ts:99]: Will add those filters and sorting as query params

File: packages/frontend/src/schema.ts (hunk 2 above)
Comment (will-sonnex) [reply packages/frontend/src/schema.ts:103]: Sure. Have a preference?

**Comment by will-sonnex** [comment]
Closing since this was a WIP to get feedback on the API design and the approach.

**Pull request closed without merging**